lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -1912,6 +1912,9 @@ namespace Js
 
         if (fParsed == TRUE)
         {
+            // Restore if the function has nameIdentifier reference, as that name on the left side will not be parsed again while deferparse.
+            funcBody->SetIsNameIdentifierRef(this->GetIsNameIdentifierRef());
```

Comment:
the bug is LHS is not an identifier ref. while I'm sure this solves the problem, is it appropriate to set name is identifier ref?  Does this bool need to be rename  to maintain spec logic in the code base?